src/MSBuild/XMake.cs

```diff
@@ -1567,6 +1567,7 @@ private static void ResetGatheringSwitchesState()
                                             nodeResultKvp.Key.ProjectInstance.GlobalProperties.All(propertyKvp => entryPoint.GlobalProperties.TryGetValue(propertyKvp.Key, out string entryValue) &&
                                                                                                                                         entryValue.Equals(propertyKvp.Value)))
                                             .Value;
+                                        success = result.OverallResult == BuildResultCode.Success;
```

Comment:
`result`, or should this use `graphResult` like the not-saving-results version and let the `result`-filtering stuff be only for extracting saved results?